integration/query_array_compat_test.go

```diff
@@ -0,0 +1,224 @@
+	"go.mongodb.org/mongo-driver/bson/primitive"
+)
+
+func TestQueryArrayCompat(t *testing.T) {
+	t.Parallel()
+
+	testCases := map[string]queryCompatTestCase{
+		"AllString": {
+			filter:        bson.D{{"v", bson.D{{"$all", bson.A{"foo"}}}}},
+			skipForTigris: "https://github.com/FerretDB/FerretDB/issues/908",
+		},
+		"AllStringRepeated": {
+			filter:        bson.D{{"v", bson.D{{"$all", bson.A{"foo", "foo", "foo"}}}}},
+			skipForTigris: "https://github.com/FerretDB/FerretDB/issues/908",
+		},
+		"AllStringEmpty": {
+			filter:        bson.D{{"v", bson.D{{"$all", bson.A{""}}}}},
+			skipForTigris: "https://github.com/FerretDB/FerretDB/issues/908",
+		},
+		"AllWhole": {
+			filter:        bson.D{{"v", bson.D{{"$all", bson.A{int32(42)}}}}},
+			skipForTigris: "https://github.com/FerretDB/FerretDB/issues/908",
+		},
+		"AllWholeNotFound": {
+			filter:     bson.D{{"v", bson.D{{"$all", bson.A{int32(44)}}}}},
+			resultType: emptyResult,
+		},
+		"AllZero": {
+			filter:        bson.D{{"v", bson.D{{"$all", bson.A{math.Copysign(0, +1)}}}}},
+			skipForTigris: "https://github.com/FerretDB/FerretDB/issues/908",
+		},
+		"AllDouble": {
+			filter:        bson.D{{"v", bson.D{{"$all", bson.A{42.13}}}}},
+			skipForTigris: "https://github.com/FerretDB/FerretDB/issues/908",
+		},
+		"AllDoubleMax": {
+			filter:        bson.D{{"v", bson.D{{"$all", bson.A{math.MaxFloat64}}}}},
+			skipForTigris: "https://github.com/FerretDB/FerretDB/issues/908",
+		},
+		"AllDoubleMin": {
+			filter:        bson.D{{"v", bson.D{{"$all", bson.A{math.SmallestNonzeroFloat64}}}}},
+			skipForTigris: "https://github.com/FerretDB/FerretDB/issues/908",
+		},
+		"AllMultiAll": {
+			filter:        bson.D{{"v", bson.D{{"$all", bson.A{"foo", 42}}}}},
+			skipForTigris: "https://github.com/FerretDB/FerretDB/issues/908",
+		},
+		"AllMultiAllWithNil": {
+			filter:        bson.D{{"v", bson.D{{"$all", bson.A{"foo", nil}}}}},
+			skipForTigris: "https://github.com/FerretDB/FerretDB/issues/908",
+		},
+		"AllEmpty": {
+			filter:     bson.D{{"v", bson.D{{"$all", bson.A{}}}}},
+			resultType: emptyResult,
+		},
+		"AllNotFound": {
+			filter:     bson.D{{"v", bson.D{{"$all", bson.A{"hello"}}}}},
+			resultType: emptyResult,
+		},
+		"$allNeedsAnArrayInt": {
+			filter:     bson.D{{"v", bson.D{{"$all", 1}}}},
+			resultType: emptyResult,
+		},
+		"$allNeedsAnArrayNil": {
+			filter:     bson.D{{"v", bson.D{{"$all", nil}}}},
+			resultType: emptyResult,
+		},
+		"DotNotationPositionIndexGreaterThanArrayLength": {
+			filter:     bson.D{{"v.5", bson.D{{"$type", "double"}}}},
+			resultType: emptyResult,
+		},
+		"DotNotationPositionIndexAtTheEndOfArray": {
+			filter:        bson.D{{"v.1", bson.D{{"$type", "double"}}}},
+			skipForTigris: "https://github.com/FerretDB/FerretDB/issues/908",
+		},
+		"DotNotationPositionTypeNull": {
+			filter:        bson.D{{"v.1", bson.D{{"$type", "double"}}}},
+			skipForTigris: "https://github.com/FerretDB/FerretDB/issues/908",
+		},
```

Comment:
What is the difference between test cases `DotNotationPositionIndexAtTheEndOfArray` and `DotNotationPositionIndexAtTheEndOfArray`?

Reply:
> `DotNotationPositionIndexAtTheEndOfArray` and `DotNotationPositionIndexAtTheEndOfArray`

That's the same name

Reply:
@AlekSi indeed! :D bad copy-pasting, fixed.

Reply:
Fixed